R-package/R/utils.R

```diff
@@ -246,3 +246,17 @@ lgb.get.default.num.threads <- function() {
     return(cores)
   }
 }
+
+lgb.equal.or.both.null <- function(a, b) {
```

Comment:
can you please add tests in https://github.com/microsoft/LightGBM/blob/master/R-package/tests/testthat/test_utils.R covering the different possible combinations of inputs here and their expected return values?

Reply:
Added.

Reply:
tests look great, thanks very much for that!